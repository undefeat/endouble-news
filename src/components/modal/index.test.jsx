import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import shallow from 'react-test-renderer/shallow';
import Modal from '.';

describe('Modal', () => {
    it('should correctly render', () => {
        const tree = shallow.createRenderer().render(
            <Modal close={() => {}}>
                <h2>Hello, World!</h2>
            </Modal>,
        );

        expect(tree).toMatchSnapshot();
    });

    it('should call "close" when close btn is clicked', () => {
        const close = jest.fn();
        const element = ReactTestUtils.renderIntoDocument(
            <Modal close={close}>
                <h2>Hello, World!</h2>
            </Modal>,
        );

        const closeButton = ReactTestUtils.findRenderedDOMComponentWithClass(element, 'modal__close-btn');
        ReactTestUtils.Simulate.click(closeButton);
        expect(close).toHaveBeenCalledTimes(1);
    });
});
