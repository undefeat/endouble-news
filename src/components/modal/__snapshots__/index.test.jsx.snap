// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should correctly render 1`] = `
Object {
  "$$typeof": Symbol(react.portal),
  "children": <div
    className="modal__wrapper"
  >
    <aside
      className="modal"
    >
      <button
        aria-label="Close"
        className="modal__close-btn"
        onClick={[Function]}
        title="Close"
        type="button"
      >
        <i
          className="fas fa-times"
        />
      </button>
      <h2>
        Hello, World!
      </h2>
    </aside>
  </div>,
  "containerInfo": <div />,
  "implementation": null,
  "key": null,
}
`;
